### PR TITLE
[AC-1121] Collections Add Access filter and badge

### PR DESCRIPTION
**Pull request by Jingo88**
```
- [ ] Bug fix
- [ X ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

- If a collection does not have any members or groups with `Can Manage` access (revoked users do not count) that collection will show an `Add Access` badge. 
- A toggle will appear at the top to filter between `All` and `Add Access` collections
- The user will be able to open the collections `Access Modal` and provide `Can Manage` access for other users, groups, and/or themselves

## Code changes

* `vault collection row html` - add badge with ngIf
* `collection admin view` - added methods to return a list of groups and users with can manage access
* `vault component html` - `bit-toggle-group` added for Add Access
* `vault.component org-vault` - When collections$ is being determined we will check for `orgRevokedUsers`, `groupsCanManage`, and `usersCanManage` and apply the badge to the appropriate collection as well as turning on the toggle group if needed

## New Screen Recording Below Along with list of all flows tested

https://github.com/bitwarden/clients/assets/8302660/2a03317f-4b55-49f4-83da-5d73e83bb88a

- Add Access Filter
	- Should show when 
		- Owners/Admin setting is off
		- Org has at least 1 orphaned Collection
	- Should only show in Collections Filter
	- Add Access Badge should not show for Unassigned collections
	- Unassigned Collections should not be filtered when using Add Access
	- If User switches between org with setting OFF to an org with setting ON the Add Access Toggle should not show
	- If a User switches to an org with Add Access on the Add Access toggle should reset
	- If viewing inside a nested Collection that has multiple collections, some orphaned and some not, along with some items. If the Add Access toggle is set to true ONLY show the orphaned collections. (NO ITEMS)
	- If Add Access Toggle is set to true. and the user navigates to a different filter such as Trash or All Items. make sure the toggle does not impact those views
- Custom Users
	- ONLY Create new - will not have permission for org
	- Edit Any Collection - Can see Edit and Access options in vault-item menu
	- ONLY Edit Any Collection - will not see Delete option in menu unless they have Can Manage access to a collection. Will not see Delete option for orphaned Collections
	- Manage All - Can see add access and edit all
	- Delete Collection - Will see delete option 
	- ONLY Delete Collection - Will see delete option for all. No Edit option unless they have Can Manage access to collection. Will not see any add access features. Will see items if those items are in a Can Manage Collection
- Owners and Admins
	- Can view Edit/Access/Delete of an orphaned Collection
	- Can open Access of an orphaned Collection and assign members to the collection, including themselves

### CODE REVIEW

**Review by LRNcardozoWDF — DISMISSED** [review]
LGTM!

File: apps/browser/src/autofill/overlay/pages/list/autofill-overlay-list.ts [hunk 1]
```diff
@@ -485,9 +487,6 @@ class AutofillOverlayList extends AutofillOverlayPageElement {
    * the first cipher button.
    */
   private focusOverlayList() {
-    this.overlayListContainer.setAttribute("role", "dialog");
-    this.overlayListContainer.setAttribute("aria-modal", "true");
-
```
Comment (cagonzalezcs) [inline apps/browser/src/autofill/overlay/pages/list/autofill-overlay-list.ts:490]: This removal shouldn't be happening as a part of your merge into `main`. These two lines are expected to be present based on a fix merged into `main` earlier this week.

Comment (cagonzalezcs) [reply apps/browser/src/autofill/overlay/pages/list/autofill-overlay-list.ts:490]: Please review https://github.com/bitwarden/clients/pull/8601 and adjust your changes for these "overlay" related files accordingly.

**Review by cagonzalezcs — CHANGES_REQUESTED** [review]

File: apps/browser/src/autofill/overlay/pages/list/autofill-overlay-list.ts (hunk 1 above)
Comment (Jingo88) [reply apps/browser/src/autofill/overlay/pages/list/autofill-overlay-list.ts:490]: yeah sry please ignore this latest comment. I accidently pushed unwanted changes up. reverting now

**Review by cagonzalezcs — APPROVED** [review]

File: apps/browser/src/autofill/overlay/pages/list/autofill-overlay-list.ts (hunk 1 above)
Comment (cagonzalezcs) [reply apps/browser/src/autofill/overlay/pages/list/autofill-overlay-list.ts]: Ah ok, no worries. Reverted my block on the PR :)

**Review by LRNcardozoWDF — DISMISSED** [review]

**Review by LRNcardozoWDF — APPROVED** [review]

**Review by LRNcardozoWDF — DISMISSED** [review]

**Review by LRNcardozoWDF — DISMISSED** [review]

**Review by LRNcardozoWDF — DISMISSED** [review]

**Review by LRNcardozoWDF — APPROVED** [review]

**Pull request merged**